src/WebJobs.Script/Workers/Profiles/WorkerProfileManager.cs

```diff
@@ -16,15 +16,13 @@ internal class WorkerProfileManager : IWorkerProfileManager
     {
         private readonly ILogger _logger;
         private readonly IEnumerable<IWorkerProfileConditionProvider> _conditionProviders;
-        private Dictionary<string, List<WorkerDescriptionProfile>> _profiles;
-        private string _activeProfile;
+        private static Dictionary<string, List<WorkerDescriptionProfile>> _profiles = new Dictionary<string, List<WorkerDescriptionProfile>>();
```

Comment:
@fabiocav any concerns with this being made static here? Or do you have a better recommendation for addressing this issue?

Reply:
This change is indeed concerning as this service is instantiated in multiple scopes and sharing this may introduce fragility and impact other people working on this code.

@shreyas-gopalakrishna we may want to consider reusing this service across different scopes instead (having a singleton instance of this service) if the multiple instances are causing problems, but this will bring additional complexities with you having to handle specialization events within the service implementation. That's not a problem, but something you need to be aware of.

Reply:
The goal here is to have 1 instance/copy of `_profiles` and `_activeProfile`. 
Is singleton the best approach we can take here? Once we have a singleton, any info on how to handle the specialization events? I recollect you pointing to AssemblyAnalysisService. Curious as to what changes to check here, is it the `IOptionsMonitor<StandbyOptions> standbyOptionsMonitor`? 

https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/AssemblyAnalysis/AssemblyAnalysisService.cs#L31